akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1114,21 +1118,30 @@ private[cluster] final class JoinSeedNodeProcess(seedNodes: immutable.IndexedSeq
  *
  * The supplied callback will be run, once, when current cluster member is `Up`.
  */
-private[cluster] class OnMemberUpListener(callback: Runnable) extends Actor with ActorLogging {
+private[cluster] class OnMemberStatusChangedListener(callback: Runnable, prefer: MemberStatus) extends Actor with ActorLogging {
   import ClusterEvent._
-  val cluster = Cluster(context.system)
-  // subscribe to MemberUp, re-subscribe when restart
+  private val cluster = Cluster(context.system)
+  private val to = prefer match {
+    case Up ⇒
+      classOf[MemberUp]
+    case Removed | Down ⇒
```

Comment:
There seem to be remnants of another feature in here as well: `Down` is not enabled for tracking by this PR.